File: Task 03/go/Subtask4/Task4.go
package main
import (
    "fmt"
    "io/ioutil"
    "strconv"
    "strings"
)

func main() {
	data, _ := ioutil.ReadFile("input.txt")
	n, _ := strconv.Atoi(strings.TrimSpace(string(data)))
	var result strings.Builder
	for i := 1; i <= n; i++ {
        for j := 0; j < n-i; j++ {
            result.WriteString(" ")
        }
        for k := 0; k < 2*i-1; k++ {
            result.WriteString("*")
        }
        result.WriteString("\n")
    }
	for i := n - 1; i >= 1; i-- {
        for j := 0; j < n-i; j++ {
            result.WriteString(" ")
        }
        for k := 0; k < 2*i-1; k++ {
            result.WriteString("*")
        }
        result.WriteString("\n")
    }
	ioutil.WriteFile("output.txt", []byte(result.String()), 0644)
}

File: Task 03/javascript/Subtask4/Task4.js
const fs = require('fs')
fs.readFile("input.txt","utf8",(err,data)=> {
    if (err){
        console.error("Error reading the file",err);
        return;

    }
    
    const n= parseInt(data.trim(),10);

    const mid= Math.floor(n/2);
    let result='';

    for(let i=0;i <=mid;i++){
        result+=' '.repeat(mid-1)+'*'.repeat(2*i+1)+'\n';

    }
    
    for(let i=mid-1;i >=0;i--){
        result+=' '.repeat(mid-i)+'*'.repeat(2*i+1)+'\n';

    }

    fs.writeFIle("output.txt",result,(err) => {
        if (err){
            console.error("Error writing the file",err);
            return;
        }
    });
});
